Sources/AsyncSequenceValidation/TaskDriver.swift

```diff
@@ -20,12 +20,12 @@ import Glibc
 #endif
 
 #if canImport(Darwin)
-func start_thread(_ raw: UnsafeMutableRawPointer) -> UnsafeMutableRawPointer? {
+@Sendable func start_thread(_ raw: UnsafeMutableRawPointer) -> UnsafeMutableRawPointer? {
```

Comment:
Adding `@Sendable` to functions isn't the right fix often. Where did the original error come from?

Reply:
This `@Sendable` is intended to address the compilation error that occurs in `TaskDriver`'s `pthread_create(&thread, nil, start_thread, Unmanaged.passRetained(self).toOpaque())` when the Swift 6 language mode is enabled.
The compilation error that occurs is as follows:

> A C function pointer can only be formed from a reference to a 'func' or a literal closure

After reconsidering based on your comments, I realized that defining `start_thread` within `start` can also suppress the compilation error.
I have tried fixing it with [ea58d5d](https://github.com/apple/swift-async-algorithms/pull/317/commits/ea58d5da0942db511abc9e8b424faec9766cc60d).